### ISSUE TITLE
GNOME shell detected, but UNKNOWN type. This should never happen. Falling back to GtkStatusIcon.

### ISSUE THREAD

**Issue by robstoll** [dorkbox/SystemTray#171]
As requested in the error message, I am creating this bug

I am using Ubuntu 20.04 with AwesomeWM
Let me know in case you need more details

**Comment by dorkbox** [comment]
Thanks!

Can you enable SystemTray.DEBUG and paste the output?

**Comment by robstoll** [comment]
```
08:45:43.249 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - OS: Linux
08:45:43.249 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Arch: amd64
08:45:43.249 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Oracle Corporation OpenJDK 64-Bit Server VM 11.0.11
08:45:43.249 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is Auto sizing tray/menu? true
08:45:43.249 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is JavaFX detected? true
08:45:43.249 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is SWT detected? false
08:45:43.307 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Java Swing L&F: Metal
08:45:43.307 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Auto-detecting tray type
08:45:43.307 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Force GTK2: false
08:45:43.307 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Prefer GTK3: true
08:45:43.643 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Currently using the 'Gnome' desktop environment
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.3 LTS"

08:45:43.644 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Currently using the 'awesome' session type
08:45:43.644 [JavaFX Application Thread] ERROR dorkbox.systemTray.SystemTray - GNOME shell detected, but UNKNOWN type. This should never happen. Falling back to GtkStatusIcon. Please create an issue with as many details as possible.
08:45:43.678 [JavaFX Application Thread] DEBUG dorkbox.jna.linux.GtkLoader - GTK: libgtk-3.so.0
08:45:43.678 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - GTK Version: 3.24.20
08:45:43.678 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is the system already running GTK? true
08:45:44.029 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Tray indicator image size: 24
08:45:44.029 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Tray menu image size: 16
08:45:44.035 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Using Fake CheckMark: false
08:45:44.038 [JavaFX Application Thread] INFO dorkbox.systemTray.SystemTray - Successfully loaded type: _GtkStatusIconNativeTray
```

**Comment by dorkbox** [comment]
awesome!

if you set `SystemTray.FORCE_TRAY_TYPE = TrayType.AppIndicator;`, does it work for you?

**Comment by robstoll** [comment]
Does not look like it:
```
Gdk-CRITICAL **: 08:59:01.873: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
```
but I guess I have a general gdk issue (I guess an update I have installed lately) and that could also be the reason why I get the first warning in the beginning

**Comment by dorkbox** [comment]
GTK/GDK is notorious for bizarre errors. Go ahead and leave this issue open - but just so you know, I won't have much time to work on this project until the holidays.

**Comment by robstoll** [comment]
@dorkbox no rush at all, the application starts as it should, even with the error above. I just figured I let you know about it as you requested it in the error message :slightly_smiling_face:

**Comment by Jeansen** [comment]
```
WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @f88832b'
22:34:35.555 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - OS: Linux
22:34:35.558 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Arch: amd64
22:34:35.558 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Oracle Corporation OpenJDK 64-Bit Server VM 17
22:34:35.558 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is Auto sizing tray/menu? true
22:34:35.558 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is JavaFX detected? true
22:34:35.558 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is SWT detected? false
22:34:35.791 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Java Swing L&F: Metal
22:34:35.791 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Auto-detecting tray type
22:34:35.793 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Force GTK2: false
22:34:35.793 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Prefer GTK3: true
22:34:35.927 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Currently using the 'Gnome' desktop environment
PRETTY_NAME="Debian GNU/Linux bookworm/sid"
NAME="Debian GNU/Linux"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

22:34:35.928 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Currently using the 'gnome-xorg' session type
22:34:35.928 [JavaFX Application Thread] ERROR dorkbox.systemTray.SystemTray - GNOME shell detected, but UNKNOWN type. This should never happen. Falling back to GtkStatusIcon. Please create an issue with as many details as possible.
22:34:36.026 [JavaFX Application Thread] DEBUG dorkbox.jna.linux.GtkLoader - GTK: libgtk-3.so.0
22:34:36.027 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - GTK Version: 3.24.30
22:34:36.027 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Is the system already running GTK? true
22:34:36.289 [JavaFX Application Thread] WARN dorkbox.jna.linux.GtkTheme - Unable to get tray image size. Using fallback: 24
22:34:36.480 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Tray indicator image size: 24
22:34:36.480 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Tray menu image size: 16
22:34:36.493 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Using Fake CheckMark: false
22:34:36.501 [JavaFX Application Thread] INFO dorkbox.systemTray.SystemTray - Successfully loaded type: _GtkStatusIconNativeTray

(java:213982): Gdk-WARNING **: 22:34:36.495: XSetErrorHandler() called with a GDK error trap pushed. Don't do that.

(java:213982): Gdk-CRITICAL **: 22:34:36.999: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
22:34:38.291 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Resizing image to 24
22:34:38.512 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Resizing image to 24
22:34:39.433 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Resizing image to 24
22:34:40.509 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Resizing image to 24
22:34:41.456 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Resizing image to 24
22:34:42.386 [JavaFX Application Thread] DEBUG dorkbox.systemTray.SystemTray - Resizing image to 24
```

**Comment by Jeansen** [comment]
> Does not look like it:
> 
> ```
> Gdk-CRITICAL **: 08:59:01.873: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
> ```
> 
> but I guess I have a general gdk issue (I guess an update I have installed lately) and that could also be the reason why I get the first warning in the beginning. I get back to you in case the problem remains once I have fixed the gdk issues
> 
> _Edit_ actually sounds like: #19 I have update to jdk17 lately. Could this be related?

I don't think so, I have this issue with different JDKs (version and flavor)

**Comment by robstoll** [comment]
@Jeansen as hint for the 
`WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @f88832b'`
We also have to deal with an unnamed module and thus added `--add-opens javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED` as parameter

Regarding jdk17, tried it out with jdk11 by now and I can confirm that the problem remains

**Comment by dorkbox** [comment]
Sadly, I'm aware of this -- and the SystemTray is one of the few remaining projects that do not support JPMS modules yet -- since it depends on eclipse projects that were made in *different* namespaces, but conflict in such a way that libraries for crossplatform use cannot be included into the same library.

I know how to fix it, but it requires a bit of time and effort to do so, and I will have the chance during the christmas holidays to fix this.

**Comment by Jeansen** [comment]
@robstoll Thanks for the hint. I will try that. @dorkbox That's good to hear/read. Thanks in advance ;-)

**Comment by Jeansen** [comment]
> @Jeansen as hint for the `WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @f88832b'` We also have to deal with an unnamed module and thus added `--add-opens javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED` as parameter
> 
> Regarding jdk17, tried it out with jdk11 by now and I can confirm that the problem remains

Yes, that worked. Then also the jar executes without any error.

**Comment by dorkbox** [comment]
Fixed bc09be72

Note: This does not fix the JMPS errors, just the AwesomeWM errors during startup,. JMPS issues are being tracked elsewhere.
